components/salsa-macros/src/query_group.rs

```diff
@@ -87,7 +87,7 @@ pub(crate) fn query_group(args: TokenStream, input: TokenStream) -> TokenStream
                 match iter.next() {
                     Some(FnArg::SelfRef(sr)) if sr.mutability.is_none() => (),
                     _ => panic!(
```

Comment:
Isn't there a way to give errors that are specific to a particular span? maybe not. It'd be nice if we could highlight the query or the `mut` keyword.

Ah, well, [I guess we could use this trick](https://internals.rust-lang.org/t/custom-error-diagnostics-with-procedural-macros-on-almost-stable-rust/8113), but that seems like a lot of work, and I wouldn't want to burden this PR with it.